File: php/src/Queue.php
<?php


namespace DataStructure;


class Queue
{
    private int $size = 0;
    private array $values = [];

    public function __construct(public $capacity)
    {
    }

    public function enqueue(mixed $value): bool
    {
        if ($this->size() >= $this->capacity)
        {
            return false;
        }
        $this->values[$this->size++] = $value;
        return true;
    }

    public function size(): int
    {
        return $this->size;
    }

    public function peek()
    {
        return 5;
    }

    public function dequeue()
    {
        $currentIndex = array_key_first($this->values);
        $value = $this->values[$currentIndex];
        unset($this->values[$currentIndex]);
        $this->size--;
        return $value;
    }
}

File: python3/queue.py
class Queue:

    def __init__(self, capacity: int):
        self.capacity = capacity
        self.__size = 0
        self.__items = []

    @property
    def size(self):
        return self.__size

    def enqueue(self, item: object) -> bool:
        if self.size >= self.capacity:
            return False
        self.__size += 1
        self.__items.append(item)
        return True

    def dequeue(self) -> object:
        if self.size < 1:
            return None
        item = self.__items[0]
        del self.__items[0]
        self.__size -= 1
        return item

    def peek(self) -> object:
        if self.size < 1:
            return None
        return self.__items[0]
